packages/snowpack/src/build/build-pipeline.ts

```diff
@@ -63,12 +68,40 @@ async function runPipelineLoadStep(
       const mainOutputExt = step.resolve.output[0];
       return {[mainOutputExt]: result};
     } else if (result && typeof result === 'object') {
+      result;
+
+      // handle source maps
+      Object.keys(result).forEach((ext) => {
+        const output = result[ext];
+        if (typeof output !== 'object' || !output.code) return;
+
+        if (output.map) {
+          // some plugins return map as an object; others as a string. handle both cases
+          result[ext + '.map'] =
+            typeof output.map === 'object' ? JSON.stringify(output.map) : output.map;
+
+          const sourceMapFile = path
```

Comment:
can we turn this into a helper, something like replaceFileExtension?

Also, see https://nodejs.org/api/path.html#path_path_basename_path_ext for a cleaner way to strip the extension